plugins/inputs/prometheus/kubernetes.go

```diff
@@ -90,7 +90,10 @@ func (p *Prometheus) startK8s(ctx context.Context) error {
 						p.Log.Errorf("Unable to monitor pods with node scrape scope: %s", err.Error())
 					}
 				} else {
-					p.watchPod(ctx, client)
+					err = p.watchPod(ctx, client)
+					if err != nil {
+						p.Log.Errorf("Unable to monitor pod: %s", err.Error())
```

Comment:
I saw this in the other PR and it made me pause. We are ignoring these errors right now, is this going to cause instances where things are working today to stop working?

Reply:
Yeah, that's fair. I wouldn't have touched this except for the linter requesting some kind of error handling, now that the underlying API returns an `error`.

I'll update the PR to log the error and continue, so that the behavior should be the same as what currently happens, with the addition of a warning message in the logs.

Reply:
Ok:

* switched to `p.Log.Warnf`, thinking that this is a non-fatal warning.
* no longer calling `<-ctx.Done()` early if `AddEventHandler` returns an error. Keeping the control flow the same as it was, with the addition of capturing the error and returning it from `watchPod`